crypto/keys/private_keys.go

```diff
@@ -159,3 +160,23 @@ func curveFromParams(params *keyspb.Specification_ECDSA) elliptic.Curve {
 	}
 	return nil
 }
+
+// MarshalPrivateKey returns an any.Any with a keyspb.PrivateKey inside.
+func MarshalPrivateKey(key crypto.Signer) (*any.Any, error) {
+	var privateKeyDER []byte
+	var err error
+
+	switch key := key.(type) {
+	case *ecdsa.PrivateKey:
+		privateKeyDER, err = x509.MarshalECPrivateKey(key)
+	case *rsa.PrivateKey:
+		privateKeyDER = x509.MarshalPKCS1PrivateKey(key)
+	default:
+		return nil, fmt.Errorf("unsupported key type: %T", key)
+	}
+	if err != nil {
```

Comment:
nit: Move `err`'s declaration and the if inside the ecdsa case? Neither need to be at this scope.